mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSource.java

```diff
@@ -173,6 +174,15 @@ public void validate(FailureCollector collector) {
     public boolean canConnect() {
       return super.canConnect() && !containsMacro(DATABASE);
     }
+
+    @Override
+    public Map<String, String> getConnectionArguments() {
+      Map<String, String> arguments = new HashMap<>(super.getConnectionArguments());
+      // If connected to MySQL > 5.0.2, and setFetchSize() > 0 on a statement,
```

Comment:
I think this should go into `getDBSpecificArguments`. Also Please add a check for fetch size. If for whatever reason it's <=0 we should not add this argument

Reply:
done